File: flutter_app/lib/store/actions/update_sensor_action.dart
import 'package:flutter_app/models/enum/pollution_levels_enum.dart';
import 'package:flutter_app/models/reading.dart';

class UpdateSensorAction {
  final Reading payload;
  final int sensorId;
  final PollutionLevels pollutionLevel;
  UpdateSensorAction(this.payload, this.sensorId, this.pollutionLevel) {}
}

File: xamarin-app/ClearSkyMaps.CP.Mobile/Store/Actions/UpdateSensorAction.cs

using ClearSkyMaps.Mobile.Models;
using ClearSkyMaps.Mobile.Models.Enums;
using Redux;
using System;
using System.Collections.Generic;
using System.Text;

namespace ClearSkyMaps.CP.Mobile.Store.Actions
{
    public class UpdateSensorAction : IAction
    {
        public Reading Payload { get; private set; }
        public int SensorId { get; private set; }
        public PollutionLevels PollutionLevel { get; private set; }
        public UpdateSensorAction(Reading payload, int sensorId, PollutionLevels pollutionLevel)
        {
            Payload = payload;
            SensorId = sensorId;
            PollutionLevel = pollutionLevel;
        }
    }
}
